### PR TITLE
PLT-7047 Add notification component

### PR DESCRIPTION
**Pull request by ben-byb**
Screenshot 2024-06-27 at 3 03 12 pm

### CODE REVIEW

File: src/components/Snackbar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  showNotification: () => {},
+});
+
+export const NotificationProvider = ({ children }: { children: ReactNode }) => {
+  const [notification, setNotification] = useState<ToastNotificationProps | null>(null);
+
+  const showNotification = useCallback((props: ToastNotificationProps) => {
+    setNotification(props);
+  }, []);
+
```
Comment (alice-byb) [inline src/components/Snackbar/index.tsx:33]: can we move notify out of the component? none of the variables it uses are from NotificationProvider

File: src/components/Snackbar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  useEffect(() => {
+    if (notification) {
+      notify(notification);
+      setNotification(null);
+    }
+  }, [notification, notify]);
+
+  return (
+    <NotificationContext.Provider value={{ showNotification }}>
```
Comment (alice-byb) [inline src/components/Snackbar/index.tsx:69]: can you put `value` behind a memo?

File: src/components/Snackbar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      alignSelf: 'center',
+    },
+  },
+
+  '.Toastify__toast-icon': {
+    height: 18,
+    width: 18,
+    borderRadius: 17,
+    padding: 4,
+    gap: 10,
```
Comment (alice-byb) [inline src/components/Snackbar/index.tsx:138]: can you change the `gap` and `padding` to be based on `theme.spacing`?

File: src/components/Snackbar/index.tsx [hunk 4]
```diff
@@ -13,7 +11,7 @@ interface ToastNotificationProps {
   type: NotificationType;
   options?: ToastOptions;
   autoClose?: number | false;
-  icon?: ToastIcon | undefined;
+  icon?: false | ((props: IconProps) => React.ReactNode) | React.ReactElement<IconProps>;
```
Comment (ben-byb) [inline src/components/Snackbar/index.tsx:14]: They dont export this type, so added manually

**Review by alice-byb — APPROVED** [review]

**Pull request merged**